packages/mask/src/plugins/Tips/SNSAdaptor/bodyViews/AddWallet.tsx

```diff
@@ -0,0 +1,175 @@
+                variant: 'error',
+                message: nowTime,
+            })
+            return
+        }
+    }, [currentPersona?.identifier, payload])
+
+    const [{ value: walletSignState }, walletSign] = useAsyncFn(async () => {
+        if (!payload || !currentPersona?.publicHexKey || !wallet.account) return false
+        try {
+            const walletSig = await Services.Ethereum.personalSign(payload.signPayload, wallet.account)
+            if (!walletSig) throw new Error('Wallet sign failed')
+            await NextIDProof.bindProof(
+                payload.uuid,
+                currentPersona.publicHexKey,
+                NextIDAction.Create,
+                NextIDPlatform.Ethereum,
+                wallet.account,
+                payload.createdAt,
+                {
+                    walletSignature: walletSig,
+                    signature: signature,
+                },
+            )
+            setSigned(true)
+            showSnackbar(t('plugin_tips_wallet_sign_success'), { variant: 'success', message: nowTime })
+            return true
+        } catch (error) {
+            showSnackbar(t('plugin_tips_wallet_sign_error'), { variant: 'error', message: nowTime })
+            return false
+        }
+    }, [currentPersona?.publicHexKey, payload, wallet, signature])
+    const [{ loading: confirmLoading, value: step = SignSteps.Ready }, handleConfirm] = useAsyncFn(async () => {
+        try {
+            if (signed) {
+                onCancel()
+                return SignSteps.SecondStepDone
+            }
+            if (!signature && !walletSignState) {
+                await personaSilentSign()
+                return SignSteps.FirstStepDone
+            } else if (signature && !walletSignState) {
+                const res = await walletSign()
+                return res ? SignSteps.SecondStepDone : SignSteps.FirstStepDone
+            } else {
+                onCancel()
+                return SignSteps.SecondStepDone
+            }
+        } catch {
+            showSnackbar('Connect error', { variant: 'error', message: nowTime })
+            return SignSteps.Ready
+        }
+    }, [signed, signature, walletSignState, walletSign, personaSilentSign])
+    const { setDialog } = useRemoteControlledDialog(WalletMessages.events.selectProviderDialogUpdated)
+    const changeWallet = useCallback(() => {
+        setDialog({
+            open: true,
+            pluginId: PluginId.Tip,
+        })
+    }, [])
+    if (!currentPersona || !wallet) return null
+
+    return (
+        <div>
+            <Steps
+                isBound={isBound}
+                notEvm={isNotEvm}
+                notConnected={!wallet.account}
+                wallet={{
+                    account: wallet.account,
+                    chainId: wallet.chainId,
+                    networkType: wallet.networkType,
+                    providerType: wallet.providerType,
+                }}
```

Comment:
```suggestion
                wallet={wallet}
```

Reply:
Use `useChainId` and `useNetworkType` in `<Steps />`.

Reply:
'<Steps />' may  handle conditions that not current connected wallet, like in popups, so i need to define its context outer the component